docs/src/components/CodeSnippet.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react'
+import Editor, { EDITOR_BACKGROUND_COLOR } from './Editor'
+
+export interface ICodeSnippetProps {
+  label?: string
+  mode?: 'jsx' | 'html' | 'sh'
+  value: string
+  style?: React.CSSProperties
+}
+
+const CodeSnippet = ({ label = '', value, mode, style, ...rest }: ICodeSnippetProps) => {
+  const editorMode: any = mode || label.replace(/.*\./, '') || 'jsx'
```

Comment:
it seems that we might obtain not supported editor mode's value here (like `tsx` from label). Is it fine to pass this value to AceEditor then, or we should opt to some default (maybe, 'text') mode, so that only explicitly supported modes will be rendered (and, thus, guaranteed to be rendered properly)?

Reply:
True.  This is partial left over from a previous iteration where `label` was named `filename`.  However, the point still stands.  I'm going to revert to using the less magical `mode` prop just as the Editor does.  This will ensure we can explicitly set the mode/highlighting separate from the label in the code snippet.

Reply:
You also asked if it is OK to pass an unsupported value, no.  It will break the highlighting and also throw.  Also, `tsx` is not a supported mode.